README.md

```diff
@@ -1,16 +1,130 @@
+* simple, small codebase
+* cross platform (linux, windows, and osx)
+* well documented and tested
+* fastest in the west
+
+
+## Downloading and Including
+
+This library is header only, so just grab the [latest release](https://github.com/babsnbabs/babs-ecs/releases) and add `#include "babs-ecs/Manager.hpp"`.
+
+If you're interested in running the tests and/or the benchmark, follow the following steps will get those built. CMake is configured to download the necessary dependencies automatically:
+1. `cd babs-ecs`
+2. `mkdir build`
+3. `cd build`
+4. `cmake ..`
+5. `make`
+
+You should then see the executables `./tests` and `./babs-benchmark` available to run.
+
+
+## Tutorial
+
+As mentioned above, this usage will be unsurprising to anyone who has used other ECS libraries. Our library is comprised of entities, components, and events.
+
+### Setting up the ECS Manager
+
+Everything revolves around an ECS Manager instance. This instance will keep track of all the entities and components, as well as provide the interface for entity searching and component retrieval.
+
+All you need to do is create an instance of the manager and you're ready to go:
+
+```c++
+#include "babs-ecs/Manager.hpp"
+
+ECS ecs;
+```
+
+### Entities
+
+Entities are easy to create and effectively just a unique identifier within the manager. It's safe to store the entities in your own systems, but make sure to go through the manager when checking the latest component availability.
+
+```c++
+#include "babs-ecs/Manager.hpp"
+
+ECS ecs;
+
+Entity entity = ecs.CreateEntity();
+entity.UUID // returns 0 since it's the first
+```
+
+### Components
+
+Components can be defined freely and registered with ECS for subsequent use. They can be as complicated as you need, so feel free to add methods.
+
+```c++
+#include <string>
+#include <iostream>
+#include "babs-ecs/Manager.hpp"
+
+struct Identity {
+    std::string name;
+};
+
+ECS ecs;
+
+ecs.RegisterComponent(Identity());
+```
+
+Once the component is registered, instances of it can be assigned to specific entities, retrieved later, and removed. An exception will be thrown if you try to access or remove component data that doesn't exist.
+
+```c++
+Entity player = ecs.CreateEntity();
+Identity player_identity{"babs"};
+
+ecs.AddComponent(player, player_identity);
+
+Identity ident = ecs.GetComponent(Identity());
+std::cout << "players name is " << ident.name << std::endl;
+
+ecs.RemoveComponent(player, Identity());
+```
+
+### Events
+
+Event systems work well with ECS for de-coupled communication between systems. Events are as easy as components to work with. These don't require registration, and you can subscribe and broadcast at any time through the event manager provided by the ECS instance.
+
+```c++
+#include "babs-ecs/Manager.hpp"
+
+ECS ecs;
+
+struct CollisionEvent
+{
+	Entity e1;
```

Comment:
Tabbing looks messed up between e1 and e2 here, can you confirm its good?

Reply:
fixed